### ISSUE TITLE
With React Native getting error "CodePush is not abstract and does not override abstract method createJSModules() in ReactPackage"

### ISSUE THREAD

**Issue by mark-friedman** [Microsoft/react-native-code-push#1139]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

### Steps to Reproduce
1. Using code-push v5.0.0-beta and React Native v0.46.3
2. Do all code-push installation steps as seen in https://github.com/Microsoft/react-native-code-push/blob/v5.0.0-beta/docs/setup-android.md
3. Run `react-native run-android`

### Expected Behavior
What you expected to happen?

Android app is built.

### Actual Behavior
What actually happens?

I get a compilation error:

```
...
:react-native-code-push:compileReleaseJavaWithJavac
:react-native-code-push:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
/Users/mark/code/thunkable/thunkable-for-ios/thunkableCompanion/node_modules/react-native-code-push/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java:21: error: CodePush is not abstract and does not override abstract method createJSModules() in ReactPackage
public class CodePush implements ReactPackage {
       ^
```

### Reproducible Demo

* Download https://github.com/Microsoft/react-native-code-push/archive/master.zip and unzip. From `Examples` folder run `node create-app.js appName react-native@0.47.1 react-native-code-push@5.0.0-beta` command to generate plain CodePushified React Native app. Please see description on top of `create-app.js` file content if needed
* If you can't reproduce the bug on it, provide us as much info as possible about your project

```
$ node create-app.js appName react-native@0.47.1 react-native-code-push@5.0.0-beta
App name: appName
React Native version: react-native@0.47.1
React Native Module for CodePush version: react-native-code-push@5.0.0-beta 

Creating CodePush app "appName-android" to release updates for android...
App "appName-android" has been created 

Deployment key for android: XCC6ZCvxfED9UqBFYWWpg1XMeDIo605fa7f6-6e43-4bb4-814a-8d23548a1849
Use "code-push release-react appName-android android" command to release updates for android 

Creating CodePush app "appName-ios" to release updates for ios...
App "appName-ios" has been created 

Deployment key for ios: 0rbrmpmwysPcAujFbOu7CZ8ljiCK605fa7f6-6e43-4bb4-814a-8d23548a1849
Use "code-push release-react appName-ios ios" command to release updates for ios 

Installing React Native...
npm WARN deprecated connect@2.30.2: connect 2.x series is deprecated
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN react-native@0.47.1 requires a peer of react@16.0.0-alpha.12 but none is installed. You must install peer dependencies yourself.

React Native has been installed 

Installing React Native Module for CodePush...
React Native Module for CodePush has been installed 

Linking React Native Module for CodePush...
React Native Module for CodePush has been linked 

fs.js:646
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open '../CodePushDemoApp/demo.js'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at setupAssets (/Users/mark/Downloads/react-native-code-push-master/Examples/create-app.js:118:36)
    at /Users/mark/Downloads/react-native-code-push-master/Examples/create-app.js:106:17
    at ChildProcess.<anonymous> (/Users/mark/Downloads/react-native-code-push-master/Examples/nexpect.js:308:9)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
```
### Environment

* react-native-code-push version: v5.0.0-beta 
* react-native version: React Native v0.46.3
* iOS/Android/Windows version: iOS 10.13.2 
* Does this reproduce on a debug build or release build?
* Does this reproduce on a simulator, or only on a physical device?

(The more info the faster we will be able to address it!)

**Comment by ruslan-bikkinin** [comment]
Hi @mark-friedman and thanks for submitting the issue. `v5.0.0-beta` of `react-native-code-push` is for react-native `v0.47.x` only. Please use `v4.1.0-beta` version of  `react-native-code-push` for RN `v0.46.3`.

Please let us know if this information was helpful for you or any updates.

**Comment by mark-friedman** [comment]
Thanks, @ruslan-bikkinin.  The `current react-native-code-push` README claims that `v5.1` will also work with React Native `v0.46-v0.52`, should I try that or is it really that only `react-native-code-push` `v4.1` that will work with React Native `v0.46.3`?

**Comment by ruslan-bikkinin** [comment]
Yes, you can use `v5.1` with RN `0.46.x` as well.

**Comment by max-mironov** [comment]
@mark-friedman ya, sorry for confusion here, we've released 5.0.0 with no backwards compatibility to RN 0.46.x, later we have released 5.1.0 which is compatible with RN 0.46 also. 
I believe we can close the issue, feel free to reach us if any questions.
